### PR TITLE
chore: bump deps

### PR DESCRIPTION
**Pull request by zirain**
fix: https://github.com/envoyproxy/gateway/pull/4103 https://github.com/envoyproxy/gateway/pull/4112

### CODE REVIEW

**Review by arkodg — DISMISSED** [review]

**Comment by zirain** [comment]
not sure why e2e always failed:
```
    --- FAIL: TestE2E/CookieBasedSessionPersistence (1.04s)
        --- FAIL: TestE2E/CookieBasedSessionPersistence/traffic_is_routed_based_on_cookie_based_session_persistence (1.01s)
        --- PASS: TestE2E/CookieBasedSessionPersistence/session_persistence_is_configured_per_route (0.01s)
```

**Review by Xunzhuo — APPROVED** [review]

File: tools/linter/golangci-lint/.golangci.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ linters:
     - copyloopvar
     - depguard
     - errorlint
-    - exportloopref
```
Comment (arkodg) [inline tools/linter/golangci-lint/.golangci.yml:10]: curious why we had to rm this

File: test/e2e/tests/envoy_shutdown.go [hunk 2]
```diff
@@ -163,7 +163,7 @@ func restartProxyAndWaitForRollout(t *testing.T, timeoutConfig config.TimeoutCon
 		}
 
 		// all pods are rolled
-		if rolled == int32(len(podList.Items)) && rolled >= *dp.Spec.Replicas {
+		if rolled == int32(len(podList.Items)) && rolled >= *dp.Spec.Replicas { // nolint: gosec
```
Comment (arkodg) [inline test/e2e/tests/envoy_shutdown.go:166]: gosec errors for type casting ?

Comment (zirain) [reply test/e2e/tests/envoy_shutdown.go:166]: Annoying warning

**Pull request merged**